### ISSUE TITLE
Clicking edge of button doesn't work

### ISSUE THREAD

**Issue by knd775** [skeletonlabs/skeleton#1789]
### Current Behavior

When clicking the very edge of a button, the animation causes the button to shrink just enough that the cursor is no longer on the button. This prevents the on click action from occuring.

![msedge_uEs2lHfYhe](https://github.com/skeletonlabs/skeleton/assets/1235055/78463681-e35f-4706-84a2-8d24c0593874)

### Expected Behavior

The click should still register and cause on click actions or navigation to occur.

### Steps To Reproduce

1. Use a button
2. Click on the edge

### More Information

I tested on Edge 114.0.1823.79 and Firefox 115.0.2

**Comment by endigo9740** [comment]
My goal will be to test this in the same browsers and on Windows so I can see this for myself. But just FYI one of the ideas I have in mind for the future is making animations opt-in rather than on by default. I think everyone enjoys the utility-based approach that Tailwind introduces here, so we'll follow that lead. It might also come with the option to exchange or interchange animations. And with the v2 version of plugin I think it will be possible to reapply these at the global level in your `app.postcss` - something that's not currently viable.

**Comment by ak4zh** [comment]
@endigo9740 Just tested this and I can confirm the issue.
I also went ahead to dig deeper into the cause and tried to remove classes one by one, first `rounded-token` then `padding` and so on. Can confirm the issue is due to `active:scale-[95%]` removing that fixed the issue.

Tested on Safari.

**Comment by endigo9740** [comment]
@knd775 @ak4zh Pinging both of you since you've both participated in this. I had a discussion around my findings for this here on Discord today:

https://discord.com/channels/1003691521280856084/1003699523522142399/1149782460889047170

Unfortunately the news isn't great. Per @ak4zh finding, it does appear that `active:scale-[95%]` is our culprit. This of course is used to add the "press" effect when you click/tap a `.btn` element.

The problem here is this is not a trivial thing to remove. While I would very much be of the mind to remove the visual effect (as we strive to keep Skeleton minimally opinionated out of the box), this would fall into the category of a breaking change. Abstracting the animation to a class `.animate-press` would not be sufficient here. Given we're changing default looks and behavior.

So we're going sit on this one for now and do the following:

- Keep an ear to the ground and try to find a potential CSS workaround. Perhaps there's additional CSS style that can be set or adjusted here that will allow the pointer events to work as expected.
- We're going to make considerations for removing or abstracting this animation out to it's own utility class. This would be a breaking change, which means it would only be possible as part of a major release of Skeleton (ex: Skeleton v3). That way it's an optional and opt-in feature if folks choose to use it. We would then put a warning in place about the browser behavior though.
- We're also going to monitor updates from browser vendors to see if they fix this, as the root issue is a browser bug. There's no reason the pointer events should match what's visible on the screen. That's clearly a bug.

I would encourage either of you to help report this browser bug upstream to the browser vendors themselves. But be aware these guys move SLOW. This could take months, if not years before they issue a fix.

We do not have an ETA for when the next major version of Skeleton might occur, but it'll likely be a ways out. Much longer than the v1 to v2 duration. My guess would be 6-12 months. Especially since a lot of changes we have planned includes new packages being generated outside the core library. We'll have little incentive to drop another major release until these changes are completed.

Long story short, this will likely be a ways out. If you absolutely need a short term fix, my recommendation is to implement this into your `app.postcss` for now:

```css
.btn { @apply !active:scale-[100%]; }
```

It's not ideal, but hopefully it can get you by for now.

**Comment by Hugos68** [comment]
Hey, after doing some testing I found a fix that utilizes the ::after pseudo element to immitate a hitbox for your cursor to click on while the animation is playing by countering the animation:
```css
.btn {
    position: relative;
}
.btn::after {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    border-radius: inherit;
    transition: inherit;
}

.btn:active::after {
    transform: scale(105%);
}
```
Stackblitz: https://stackblitz.com/edit/skeletonlabs-repl-zfqzkq?file=src%2Froutes%2F%2Bpage.svelte
Note: This was done and tested *only* in the v1 stackblitz, I would expect to work fine in v2 but I have not tested this yet.

**Comment by endigo9740** [comment]
In an effort to prepare for Skeleton v3, we're consolidate some related issues down to a single ticket. This will ensure that we can see the full context of requests when the time comes to refactor and update this feature going forward. If you wish to add additional feedback or suggestions, please so here:

- https://github.com/skeletonlabs/skeleton/issues/2360
